api/pkg/gateway/context_test.go

```diff
@@ -0,0 +1,541 @@
+package gateway
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/shellhub-io/shellhub/pkg/api/authorizer"
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRole(t *testing.T) {
+	type Expected struct {
+		roleString authorizer.Role
+	}
+
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    Expected
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Role": "owner",
+			},
+			expected: Expected{
+				roleString: authorizer.Role("owner"),
+			},
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Role": "",
+			},
+			expected: Expected{
+				roleString: authorizer.Role(""),
+			},
+		}, {
+			description: "validate empty headers, for fail function",
+			headers: map[string]string{
+				"": "",
+			},
+			expected: Expected{
+				roleString: authorizer.Role(""),
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodPost, "/", nil)
+
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			for k, v := range tc.headers {
+				req.Header.Set(k, v)
+			}
+
+			rec := httptest.NewRecorder()
+			ctxEchoT := e.NewContext(req, rec)
+
+			gatewayCtxT := Context{
+				nil,
+				ctxEchoT,
+			}
+
+			retRoleString := gatewayCtxT.Role()
+
+			require.Equal(t, tc.expected.roleString, retRoleString)
+		})
+	}
+}
+
+func TestTenant(t *testing.T) {
+	type Expected struct {
+		tenantID *models.Tenant
+	}
+
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    Expected
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Tenant-ID": "00000000-0000-4000-0000-000000000000",
+			},
+			expected: Expected{
+				tenantID: &models.Tenant{
+					ID: "00000000-0000-4000-0000-000000000000",
+				},
+			},
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Tenant-ID": "",
+			},
+			expected: Expected{
+				tenantID: nil,
+			},
+		}, {
+			description: "validate empty headers, for fail function",
+			headers: map[string]string{
+				"": "",
+			},
+			expected: Expected{
+				tenantID: nil,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodPost, "/", nil)
+
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			for k, v := range tc.headers {
+				req.Header.Set(k, v)
+			}
+
+			rec := httptest.NewRecorder()
+			ctxEchoT := e.NewContext(req, rec)
+
+			gatewayCtxT := Context{
+				nil,
+				ctxEchoT,
+			}
+
+			retTenantID := gatewayCtxT.Tenant()
```

Comment:
As mentioned by @heiytor, the `ret` prefix isn't necessary and doesn't follow the Golang pattern for variables in general. I believe it could be simple `tenant` or `tenantID` without cognitive prejudice.

You could check these pattern here: https://go.dev/doc/effective_go#variables